client/src/components/Donation/donation-modal-body.tsx

```diff
@@ -0,0 +1,263 @@
+  showForm,
+  donationAnimationFlag
+}: {
+  recentlyClaimedBlock: RecentlyClaimedBlock;
+  showHeaderAndFooter: boolean;
+  donationAttempted: boolean;
+  showForm: boolean;
+  donationAnimationFlag: boolean;
+}) {
+  const { t } = useTranslation();
+  if (showHeaderAndFooter && !donationAttempted) {
+    if (!showForm && donationAnimationFlag) {
+      return (
+        <Row className='text-center block-modal-text'>
+          <Col sm={10} smOffset={1} xs={12}>
+            <h2>{t('donate.modal-benefits-title')}</h2>
+          </Col>
+        </Row>
+      );
+    }
+    if (!donationAnimationFlag)
+      return (
+        <Row className='text-center block-modal-text'>
+          <Col sm={10} smOffset={1} xs={12}>
+            {recentlyClaimedBlock !== null && (
+              <b>
+                {t('donate.nicely-done', {
+                  block: t(
+                    `intro:${recentlyClaimedBlock.superBlock}.blocks.${recentlyClaimedBlock.block}.title`
+                  )
+                })}
+              </b>
+            )}
+            <h2>{t('donate.help-us-develop')}</h2>
+          </Col>
+        </Row>
+      );
+  }
+
+  return null;
+}
+
+function CloseButtonRow({
+  donationAttempted,
+  closeDonationModal
+}: {
+  donationAttempted: boolean;
+  closeDonationModal: () => void;
+}) {
+  const { t } = useTranslation();
+  return (
+    <Row>
+      <Col sm={4} smOffset={4} xs={8} xsOffset={2}>
+        <button
+          className='close-button'
+          type='button'
+          onClick={closeDonationModal}
+        >
+          {donationAttempted ? t('buttons.close') : t('buttons.ask-later')}
+        </button>
+      </Col>
+    </Row>
+  );
+}
+
+const Benefits = ({
+  setShowForm,
+  executeGA
+}: {
+  setShowForm: (arg: boolean) => void;
+  executeGA: (arg: { event: string; action: string }) => void;
+}) => {
+  const { t } = useTranslation();
+
+  const handleBecomeSupprterClick = () => {
+    executeGA({
+      event: 'donation_related',
+      action: `Modal Become Supporter Click`
+    });
+    setShowForm(true);
+  };
+  return (
+    <Row className={'donate-btn-group'}>
+      <Col xs={12}>
+        <ModalBenefitList />
+        <Spacer size='small' />
+        <button
+          className='text-center confirm-donation-btn donate-btn-group'
+          type='submit'
+          onClick={handleBecomeSupprterClick}
+        >
+          {t('donate.become-supporter')}
+        </button>
+        <Spacer size='medium' />
+      </Col>
+    </Row>
+  );
+};
+
+function DonationModalBody({
+  closeDonationModal,
+  recentlyClaimedBlock,
+  executeGA
+}: DonationModalBodyProps): JSX.Element {
+  const [donationAttempted, setDonationAttempted] = useState(false);
+  const [showHeaderAndFooter, setShowHeaderAndFooter] = useState(true);
+  const donationAnimationFlag = useFeature('donation-animation').on;
+
+  // uncomment to testing the animation
+  // recentlyClaimedBlock = true;
+  // donationAnimationFlag = true;
```

Comment:
```suggestion
  // uncomment to testing the animation
  // recentlyClaimedBlock = true;
  // donationAnimationFlag = true;
```
Uncomment  donationAnimationFlag to test the animation.
Uncomment recentlyClaimedBlock to see how the modal will look on block completion.